packages/core/src/client/bundlerClient.ts

```diff
@@ -79,20 +79,20 @@ export function createBundlerClient(
         fetchOptions: HttpTransportConfig["fetchOptions"];
         url: string;
       };
+      if (fetchOptions) {
+        fetchOptions.headers = {
+          ...fetchOptions.headers,
+          ...(url.toLowerCase().indexOf("alchemy") > -1
+            ? { "Alchemy-AA-Sdk-Version": VERSION }
+            : undefined),
+        };
```

Comment:
(Nit) This looks a little odd in that copying the object would normally be done to avoid mutating the input, but here we're mutating the input anyways by setting `fetchOptions.headers` to a different value.

If you don't mind mutating the input, this might be a little easier to read as
```ts
if (fetchOptions && url.toLowerCase().indexOf("alchemy") > -1) {
  fetchOptions.headers["Alchemy-AA-Sdk-Version"] = VERSION;
}
```

Reply:
ah ok so I tried this out. if I do it the way you've got it here, I get a TS error that `HeadersInit` doesn't have a property `"Alchemy-AA-Sdk-Version"` so I can't index directly into it this way. So I still need to use the spread in the reassignment. I've cleaned it up though so it's much easier to read

Reply:
I think that type error might be highlighting a real bug, actually. `fetchOptions.headers` is type `HeadersInit`, which means it could also be a list of key-value pairs or an instance of the `Headers` class, which would make this code not work.